SpringParking-parkingBranch/src/test/java/pl/springDemo/demo/userPaymentTest/UserPaymentRepoTest.java

```diff
@@ -0,0 +1,57 @@
+package pl.springDemo.demo.userPaymentTest;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.test.context.junit4.SpringRunner;
+import pl.springDemo.demo.user.UserType;
+import pl.springDemo.demo.userPayment.UserPayment;
+import pl.springDemo.demo.userPayment.UserPaymentRepo;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class UserPaymentRepoTest {
+
+    @Autowired
+    private TestEntityManager entityManager;
+
+    @Autowired
+    private UserPaymentRepo userPaymentRepo;
+
+    @Test
+    public void whenFindDateFrom_thenReturnUserPayment(){
+
+        //given
+        UserPayment userPayment = new UserPayment(1,LocalDateTime.now(),LocalDateTime.now().plusHours(1),LocalDate.now(),1,UserType.DRIVER);
+        entityManager.persist(userPayment);
```

Comment:
czemu nie wyrzucić powtarzających się części given we wspólne miejsce?

Reply:
Jak najbardziej mogłabym uniknąć powtarzającego się kodu. Mimo to pomyślałam, że przy tak niewielu metodach testowych mogę to zostawić, jednak zgadzam się, że choćby umieszczenie powtarzających się części given (obiektów userPayment) w sekcji @Before zdecydowanie zaoszczędziło by to wielu nadmiarowych linijek powtarzającego się kodu i byłby on bardziej przejrzysty.